python/setup.py

```diff
@@ -1,11 +1,41 @@
 import os
+import codecs
 import sys
 import shutil
 import subprocess
 from pathlib import Path
 
 from setuptools import Extension, setup
 from setuptools.command.build_ext import build_ext
+import setuptools_scm as scm
+import setuptools_scm.git as scmgit
+
+def get_git_revision_short_hash() -> str:
+    try:
+        sha = subprocess.check_output(['git', 'rev-parse', '--short', 'HEAD']).decode('ascii').strip()
+        return f"+{sha}"
+    except Exception:
+        return ''
+
+def read(rel_path):
+    here = os.path.abspath(os.path.dirname(__file__))
+    with codecs.open(os.path.join(here, rel_path), 'r') as fp:
+        return fp.read()
+
+def get_version(rel_path):
+    new_ver = None
+    with open(rel_path) as f:  lines = f.read().splitlines()
+    with open(rel_path, 'w') as f:
+        for line in lines:
+            if line.startswith('__version__'):
+                delim = '"' if '"' in line else "'"
+                new_ver = line.split(delim)[1].split("+")[0] + get_git_revision_short_hash()
+                f.write(line.replace(line.split(delim)[1], new_ver))
```

Comment:
You can ask `setuptools_scm` to write version file using `version_file` and `version_file_template`.